tupa/parse.py

```diff
@@ -644,6 +644,28 @@ def read_passages(args, files):
                                       converters=CONVERTERS, lang=Config().args.lang)
 
 
+def filter_passages_for_bert(passages, args):
```

Comment:
Shouldn't this return a generator? In case we don't want to load all files to memory before starting to train/parse.

Reply:
Generator doesn't work with the shuffle function.
I agree its not optimal and should maybe be fixed in the future (or now if you have a better idea :) )

Reply:
Oh, with a [LazyLoadedPassages](https://github.com/danielhers/ucca/blob/master/ucca/ioutil.py#L21) object the shuffle will shuffle the filenames and still the loading of the actual passages is lazy.

Reply:
Can you confirm that doing the filtering in Parser.train() and Parser.parse() will cover all the code paths?

Reply:
I think it should be done together with `annotate_all`. Then it will cover all paths.

Reply:
annotate_all is done only in Parser.parse() function. I will do it therre

Reply:
Right. Since train() calls parse(), it will work.

Reply:
Great, now just need to yield instead of return a list :)

Reply:
Oh right, forgot about it ^^' Good tihng you are taking a second look